src/plugins/vis_type_pie/public/pie_component.tsx

```diff
@@ -0,0 +1,338 @@
+import { ChartSplit, SMALL_MULTIPLES_ID } from './components/chart_split';
+
+import './chart.scss';
+
+declare global {
+  interface Window {
+    /**
+     * Flag used to enable debugState on elastic charts
+     */
+    _echDebugStateFlag?: boolean;
+  }
+}
+
+export interface PieComponentProps {
+  visParams: PieVisParams;
+  visData: Datatable;
+  uiState: PersistedState;
+  fireEvent: IInterpreterRenderHandlers['event'];
+  renderComplete: IInterpreterRenderHandlers['done'];
+  chartsThemeService: ChartsPluginSetup['theme'];
+  palettesRegistry: PaletteRegistry;
+  services: DataPublicPluginStart;
+  syncColors: boolean;
+}
+
+const PieComponent = (props: PieComponentProps) => {
+  const chartTheme = props.chartsThemeService.useChartsTheme();
+  const chartBaseTheme = props.chartsThemeService.useChartsBaseTheme();
+  const [showLegend, setShowLegend] = useState<boolean>(() => {
+    const bwcLegendStateDefault =
+      props.visParams.addLegend == null ? false : props.visParams.addLegend;
+    return props.uiState?.get('vis.legendOpen', bwcLegendStateDefault) as boolean;
+  });
+
+  const onRenderChange = useCallback<RenderChangeListener>(
+    (isRendered) => {
+      if (isRendered) {
+        props.renderComplete();
+      }
+    },
+    [props]
+  );
+
+  // handles slice click event
+  const handleSliceClick = useCallback(
+    (
+      clickedLayers: LayerValue[],
+      bucketColumns: Array<Partial<BucketColumns>>,
+      visData: Datatable,
+      splitChartDimension?: DatatableColumn,
+      splitChartFormatter?: FieldFormat
+    ): void => {
+      const data = getFilterClickData(
+        clickedLayers,
+        bucketColumns,
+        visData,
+        splitChartDimension,
+        splitChartFormatter
+      );
+      const event = {
+        name: 'filterBucket',
+        data: { data },
+      };
+      props.fireEvent(event);
+    },
+    [props]
+  );
+
+  // handles legend action event data
+  const getLegendActionEventData = useCallback(
+    (visData: Datatable) => (series: SeriesIdentifier): ClickTriggerEvent | null => {
+      const data = getFilterEventData(visData, series);
+
+      return {
+        name: 'filterBucket',
+        data: {
+          negate: false,
+          data,
+        },
+      };
+    },
+    []
+  );
+
+  const handleLegendAction = useCallback(
+    (event: ClickTriggerEvent, negate = false) => {
+      props.fireEvent({
+        ...event,
+        data: {
+          ...event.data,
+          negate,
+        },
+      });
+    },
+    [props]
+  );
+
+  const toggleLegend = useCallback(() => {
+    setShowLegend((value) => {
+      const newValue = !value;
+      props.uiState?.set('vis.legendOpen', newValue);
+      return newValue;
+    });
+  }, [props.uiState]);
+
+  useEffect(() => {
+    setShowLegend(props.visParams.addLegend);
+    props.uiState?.set('vis.legendOpen', props.visParams.addLegend);
+  }, [props.uiState, props.visParams.addLegend]);
+
+  const setColor = useCallback(
```

Comment:
Not sure where to put this comment, but the custom colors are getting lost in the following scenario:

1. Set a custom color
2. Change the palette
3. Click the color icon that has a custom color

Reply:
Can you give me more info on this? I can't replicate it.